Review request for To Do list: add & remove

mercymugambi opened the pull request:
## On this milestone

- Removed all hardcoded items from the tasks array created in first project.
- Created a new JavaScript file for the new functionality.
- Implemented a function for adding a new task (add a new element to the array).
- Implemented a function for deleting a task (remove an element from the array).
- Implemented a function for editing task descriptions.
- By default new tasks acquire property completed set to false and the property index set to the value of the new array length.
- Deleting a task, update all remaining items' indexes, so they represent the current list order and are unique(i.e. if you're deleting the first task index 1 from the list, the index of the next task(2) is set to 1)..
- All changes to the To Do List are saved in local storage.

mercymugambi committed:
Install the webpack

mercymugambi committed:
Add content to js file and install linters

mercymugambi committed:
Fix linters

mercymugambi committed:
fix dist files build

mercymugambi committed:
Fix input type to add items

mercymugambi committed:
Fix local storage and indexing of items

mercymugambi committed:
Run dist build

OLIPLICHE reviewed (commented):
Hi @mercymugambi, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

Hi, I would advise you to use the [ReadMe template](https://github.com/microverseinc/readme-template) provided by Microverse. Kindly work on the ReadMe accordingly to avoid more invalidation of your project. A clean ReadMe with details in it will make your project looks more professional.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

mercymugambi committed:
Update Readme

DammyShittu commented:
Hi @mercymugambi,

While you made a great effort in this project, I cannot proceed to review your code.

## Invalid Code Review Request

### Most parts of your readme file are describing Portfolio project from the HTML module. Kindly note that we have moved on from that and so the readme should be updated accordingly based on the project. Kindly edit your readme to fit this project.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

mercymugambi committed:
Update readme

mercymugambi committed:
Update readme

mercymugambi committed:
Update README.md

mercymugambi committed:
Update README.md

nahnahdev commented on src/print.js:
```diff
@@ -0,0 +1,33 @@
+
+  Element.innerHTML = finalHtmlItem;
+
+  // Add event listener to each remove icon
+  const removeIcons = document.querySelectorAll('.remove-icon');
+  removeIcons.forEach((icon) => {
+    icon.addEventListener('click', (event) => {
+      const li = event.target.closest('li');
+      const index = parseInt(li.dataset.index, 10); // Specify radix 10 to avoid warnings
+      const checkbox = li.querySelector('.Checkboxi');
+      if (checkbox.checked) {
+        Items.splice(index - 1, 1); // Removes the corresponding element from the array
+        AddItems(Items, Element); // Re-render the list
+      }
```
- [x] Good job so far @mercymugambi  However, Your delete functionality is not working as expected. After adding a to-do item, I am unable to delete it. I have attached a video reference below. :wink: 
- Getting to the end of the video after clicking on the icon repeatedly. The first to-do item was deleted but the others didn't. I noticed that it was still in the local storage even though it was deleted on the screen so after refreshing the page, it was readded. :thinking: 
- So kindly ensure that the delete functionality work as expected and that deleting a task should update all remaining items' indexes, so they represent the current list order and are unique(i.e. if you're deleting the first task index 1 from the list, the index of the next task(2) should set to 1. :+1: 
- [x] Please ensure that all these updates are saved to the local storage as specified in the [ Project requirements](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m3_add_remove.md) :100: 
- [x] I also noticed that you don't have a function for editing a to-do item. Kindly include this as well since in the[ Project requirements](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m3_add_remove.md), you were asked to Implement a function for editing task descriptions. :+1: 

 - Video Reference
 
[Screencast from 03-31-2023 11:14:04 AM.webm](https://user-images.githubusercontent.com/102699958/229107130-65293e53-1109-4156-b81b-0351654321d2.webm)

nahnahdev reviewed (changes requested):
## Project Status: CHANGES REQUIRED :recycle: 
Hi @mercymugambi , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<img src="https://media4.giphy.com/media/neUaH5fz6ZpefE2KBP/giphy.gif"/>

## Highlights
- Add functionality works as expected  :heavy_check_mark: 
- No linter errors :heavy_check_mark: 
- Professional Readme :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

mercymugambi committed:
Update description function to save on local storage

mercymugambi committed:
Merge branch 'add-remove-item' of https://github.com/mercymugambi/to-do-list-webpack-project into add-remove-item

mercymugambi committed:
Update description function to save on local storage

mercymugambi committed:
Fix linters error

mercymugambi commented:
## Changes implemented as requested

- [x] Fixed delete functionality to delete items.
- [x] Fixed description to be editable and updated on local storage.
- [x] Fixed index to update after an item is deleted.

huseyinbiyik commented on src/index.html:
```diff
@@ -0,0 +1,36 @@
+            <div class="add-items">
+                <input type="text" name="add-items" placeholder="Add to your list..." id="add-item" class="addItem"/>
+                <i id="icon" class='fas fa-level-down-alt image'></i>
+                <!-- <span class="icon">&#x1F50D;</span> -->
```
[OPTIONAL] Please don't leave any uncommented code in your codebase. In this way, you can keep your codebase clean, organized and professional. If you need a specific code you can use git history. 👍

huseyinbiyik commented on src/print.js:
```diff
@@ -0,0 +1,41 @@
+
+  // Add event listener to each remove icon
+  const removeIcons = document.querySelectorAll('.remove-icon');
+  removeIcons.forEach((icon) => {
```
Adding and editing works well 👍  But let's fix the remove functionality of our tasks as the previous reviewer mentioned as well. Currently there is no action when clicking to the button on the right:
![l](https://user-images.githubusercontent.com/25887253/229212333-178e2883-6e41-4789-9358-18963c113a7c.gif)

huseyinbiyik reviewed (changes requested):
Hi @mercymugambi 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
![tenor](https://user-images.githubusercontent.com/25887253/205415161-00e79d1b-af68-4f10-9b2c-265f460dc5da.gif)

 ## Highlights
- Add and edit functionality works fine 👍 
- Linter set up well and checks pass 🟢
- Github flow followed 👌
- PR description is descriptive 📝


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

mercymugambi committed:
Add event listener to display trash and description text-decoration

mercymugambi committed:
Fix drag and drop for item/update index on local storage

mercymugambi committed:
Remove deadlinks

codepantha reviewed (approved):
# STATUS: APPROVED 🟢 ✔️ 

## Hi @mercymugambi 👋, 

Well done for implementing the required changes 👏 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

<img src="https://media3.giphy.com/media/2RGhmKXcl0ViM/giphy.gif"/>

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
